pkg/controller/statusmanager/status_manager.go

```diff
@@ -158,15 +159,32 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...condi
 			},
 		)
 	} else if reachedAvailableLevel {
-		// If successfully deployed all components and is not failing on anything, mark as Available
-		status.eventEmitter.EmitAvailableForConfig()
-		conditionsv1.SetStatusConditionNoHeartbeat(&config.Status.Conditions,
-			conditionsv1.Condition{
-				Type:   conditionsv1.ConditionAvailable,
-				Status: corev1.ConditionTrue,
-			},
-		)
-		config.Status.ObservedVersion = operatorVersion
+		if status.isRunningOnOpenshift411OrLater() && config.Spec.NMState != nil {
+			// In case the clusterVersion.Status.Desired.Version is 4.11 and we're deploying nmstate
+			// with CNAO, we'll set degraded state to block CNV upgrade, forcing migration of
+			// nmstate to kubernetes nmstate operator
+			reason := "InvalidConfiguration"
+			message := "NMState deployment is not provided in CNV >= 4.11, please install Kubernetes NMState Operator"
+			status.eventEmitter.EmitFailingForConfig(reason, message)
+			conditionsv1.SetStatusConditionNoHeartbeat(&config.Status.Conditions,
```

Comment:
This should change heartbeat right ? it's not a noop

Reply:
Right correct